frontend/src/dataset_explorer/TableChartDataTable.tsx

```diff
@@ -0,0 +1,150 @@
+
+/** Corollary for TableHeader **/
+function TableHeader({ column }: { column: HeaderGroup<any> }) {
+  return (
+      <TableCell {...column.getHeaderProps(column.getSortByToggleProps())}>
+        {column.render("Header")}
+        <TableSortLabel
+            active={column.isSorted}
+            direction={column.isSortedDesc ? "desc" : "asc"}
+        />
+      </TableCell>
+  );
+}
+
+/** Corollary for TableHeaderGroup **/
+function TableHeaderGroup({ group }: { group: HeaderGroup<any> }) {
+  return (
+      <TableRow {...group.getHeaderGroupProps()}>
+        {group.headers.map((col, index) => (
+            <TableHeader column={col} key={index} />
+        ))}
+      </TableRow>
+  );
+}
+
+export interface DataTableProps {
+  data: Readonly<Record<string, any>>[];
+  metrics: MetricConfig[];
+  breakdownVar: BreakdownVar;
+}
+
+function TableChartDataTable(props: DataTableProps) {
+  const { data, metrics, breakdownVar } = props;
+  let columns = metrics.map((metricConfig) => {
+    return {
+      Header: metricConfig.fullCardTitleName,
+      accessor: metricConfig.metricId,
+    };
+  });
+  columns = [{
+    Header: BREAKDOWN_VAR_DISPLAY_NAMES[breakdownVar],
+    accessor: breakdownVar as MetricId,
+  }].concat(columns);
+
+  const memoCols = React.useMemo<Column<any>[]>(() => columns, []);
+  const memoData = React.useMemo(() => data, [data]);
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    page,
+    prepareRow,
+    gotoPage,
+    setPageSize,
+    state: {pageIndex, pageSize},
+  } = useTable(
+      {
+        columns: memoCols,
+        data: memoData,
+        initialState: {pageSize: 5},
+      },
+      useSortBy,
+      usePagination
+  );
+
+  function CustomTableRow({ row, }: { row: Row<any> }) {
+    prepareRow(row);
+    return (
+        <TableRow {...row.getRowProps()}>
+          {row.cells.map((cell, index) => (
+              <CustomTableCell cell={cell} key={index} />
+          ))}
+        </TableRow>
+    );
+  }
+
+  /** Corollary for CustomTableCell **/
+  function CustomTableCell({ cell }: { cell: Cell<any> }) {
+    if (cell.value == null) {
+      return (
+          <TableCell {...cell.getCellProps()}>
+            <Tooltip title="No data available">
+              <WarningRoundedIcon />
+            </Tooltip>
+          </TableCell>
+      );
+    }
+    // TODO(efregoso): This could be iffy -- need to use formatFieldValue from MetrigConfig
+    if (cell.column.id.includes("_pct")) {
+      return <TableCell {...cell.getCellProps()}>{cell.render("Cell")}%</TableCell>;
+    }
+    else {
+      return <TableCell {...cell.getCellProps()}>{cell.render("Cell")}</TableCell>;
+    }
+  }
+
+  return (
+      <TableContainer component={Paper} style={{maxHeight: "500px"}}>
+        <Table stickyHeader {...getTableProps()}>
+          <TableHead>
+            {headerGroups.map((group, index) => (
+                <TableHeaderGroup group={group} key={index} />
```

Comment:
Just a style preference, but I might rename this and CustomTableRow such that their names echo their distinct and mutually exclusive roles more explicitly: TableHeaderRow and TableDataRow